token-lending/program/src/processor.rs

```diff
@@ -1714,6 +1719,85 @@ fn unpack_mint(data: &[u8]) -> Result<Mint, LendingError> {
     Mint::unpack(data).map_err(|_| LendingError::InvalidTokenMint)
 }
 
+fn get_pyth_product_quote_currency(pyth_product: &pyth::Product) -> [u8; 32] {
+    let mut quote_currency = [0u8; 32];
+
+    let mut attribute_iter = pyth_product.attr[..].iter();
```

Comment:
Iterating over each byte is probably too inefficient. It would be better to use indexes